File: AtCoder/ACP/BootCamp4b/Medium100_2/python/C_When_I_hit_my_pocket.py
K, A, B = map(int, input().split())
if K <= A - 1 or B <= A:
    res = K + 1
else:
    k = K - (A - 1)
    res = max(K + 1, A + (B - A) * (k // 2) + k % 2)
print(res)

File: AtCoder/ACP/BootCamp4b/Medium100_2/cpp/C_When_I_hit_my_pocket.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

ll K, A, B;

void solve() {
  ll res;
  if (K <= A - 1 || B <= A) {
    res = K + 1;
  } else {
    ll k = K - (A - 1);
    res = max(K + 1, A + (B - A) * (k / 2) + k % 2);
  }
  printf("%lld\n", res);
}

int main() {
  cin >> K >> A >> B;
  solve();
}